lib/type1.g

```diff
@@ -231,6 +221,7 @@ BIND_GLOBAL( "NEW_TYPE", function ( typeOfTypes, family, flags, data )
         cached := cache[hash];
         if IS_EQUAL_FLAGS( flags, cached![2] )  then
             if    IS_IDENTICAL_OBJ(  data,  cached![ POS_DATA_TYPE ] )
+              and IS_IDENTICAL_OBJ( parent, cached![ POS_PARENT_TYPE] )
```

Comment:
Some questions and thoughts:

1. I wonder if this is ever hit (that is: how often there is a cache hit where data and typeOfTypes match, but the parent does not). And if it is, then how often, and under which circumstances. I think knowing that would grant us some importan insights. So perhaps we should modify the code to (optionally) log this information, say if the assertion level is high enoug?

2. How many types are there left that add stuff beyond POS_DATA_TYPE? Doing that seems quite hard with this patch, no? Perhaps I am blind, but the only way I can think of is to somehow end up calling NEW_TYPE with a "parent" this is not actually a type (i.e. was "hand made"). But NewType5 is never used (as per my analysis in #157), and SupType / Subtype cannot be induced to do this, I think. 
If this is true, we can perhaps simplify the code again, and get rid of POS_PARENT_TYPE?

3. OTOH. strictly speaking, maybe even this check is not quite enough, because later on, we copy the content of the parent type into the new type. But between the creation of the cached type, and now, the parent type could have been modified, couldn't it? Or is the "readonly" property fully recursive, and ensures that subobject are truly immutable? If so, ignore me. If not, we could be doubly paranoid, and verify (only if the assertion level is high enough, of course) that the entries of the parent type and the cached type are identical). If we follow that thought to its conclusion: Can't we then avoid adding `POS_PARENT_TYPE`, and instead compare the trailing type members? (If they even exist, see the previous point)